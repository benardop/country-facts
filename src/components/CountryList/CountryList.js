import React from 'react';
import PropTypes from 'prop-types';
import CountryItem from '../CountryItem/CountryItem';
import style from './style.module.css';

function CountryList(props) {
  const { countries } = props;
  return (
    <div className={style.container}>
      {countries.map((el, index) => <CountryItem country={el} key={el.name} st={index % 2 === 0 ? 'rgb(65,104,177)' : 'rgb(63,98,166)'} />)}
    </div>
  );
}
CountryList.propTypes = {
  countries: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export default CountryList;
