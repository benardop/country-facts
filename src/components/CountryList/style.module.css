.container {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    background-color: rgb(53, 84, 139);
    box-sizing: border-box;
    padding: 1rem;
  }
  