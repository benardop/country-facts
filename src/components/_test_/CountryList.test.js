import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CountryList from '../CountryList/CountryList';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

const countries = [
  {
    name: 'Iraq',
    capital: 'Bagdad',
    flag: 'https//flag',
    timezone: 'utc+1',
  },
  {
    name: 'Iraq',
    capital: 'Bagdad',
    flag: 'https//flag',
    timezone: 'utc+1',
  },
];

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CountryList countries={countries} />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryList countries={countries} />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('countryList-container');
  expect(container).toBeInTheDocument();
});
