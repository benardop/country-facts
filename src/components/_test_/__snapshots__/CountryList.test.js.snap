// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="container"
      style={
        Object {
          "backgroundColor": "rgb(65,104,177)",
        }
      }
    >
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <img
          alt="arrow"
          src="arrow.svg"
        />
      </a>
      <h1>
        Iraq
      </h1>
      <div>
        <img
          alt="flag"
          onError={[Function]}
          src="https//flag"
        />
      </div>
      <h2>
        Capital :
        Bagdad
      </h2>
      <h2>
        Timezone :
        <br />
         
        utc+1
      </h2>
    </div>
    <div
      className="container"
      style={
        Object {
          "backgroundColor": "rgb(63,98,166)",
        }
      }
    >
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <img
          alt="arrow"
          src="arrow.svg"
        />
      </a>
      <h1>
        Iraq
      </h1>
      <div>
        <img
          alt="flag"
          onError={[Function]}
          src="https//flag"
        />
      </div>
      <h2>
        Capital :
        Bagdad
      </h2>
      <h2>
        Timezone :
        <br />
         
        utc+1
      </h2>
    </div>
  </div>,
  ",",
]
`;
